MoreLinq/Create.cs

```diff
@@ -0,0 +1,48 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2017 Atif Aziz. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Creates an <see cref="IEnumerable{T}"/> given a factory function
+        /// for an <see cref="IEnumerator{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">Type of returned sequence elements.</typeparam>
+        /// <param name="factory">
+        /// Function that creates an <see cref="IEnumerator{T}"/>.</param>
+        /// <remarks>
+        /// This method uses deferred execution semantics and streams its
+        /// results.
+        /// </remarks>
+
+        public static IEnumerable<T> Create<T>(Func<IEnumerator<T>> factory)
```

Comment:
> I agree, it's arguable and I wouldn't be completely upset if we drop it. I had plans to build a couple of operators on top of it but have scrapped those.

OOC, what other operators? If we have something that builds on top of `Create`, it would be weird if we didn't provide it.

> The only advantage of having it in MoreLINQ, however, is that System.Interactive targets .NET 4.5 as a minimum, which is a lot to ask for something as trivial as Create.

Well, it's not like 4.5 is such a hard requirement to make. It has been around for 5 years already and can be installed [on any currently supported OS](https://www.microsoft.com/en-us/download/details.aspx?id=42642). 3.5 SP1 is the [only older version that is still supported](https://support.microsoft.com/en-us/lifecycle/search?alpha=.net%20framework), and AFAICT, it is only offered for [old OS](https://www.microsoft.com/en-us/download/details.aspx?id=22), which are already only [on extended support](https://support.microsoft.com/en-us/lifecycle/search?sort=PN&alpha=Windows%20Server%202008&Filter=FilterNO).

I wonder if we should just bite the bullet and do the same for MoreLinq. It would save [having to create single-purpose, internal structs](https://github.com/morelinq/MoreLINQ/pull/287/files#diff-d45d5bd92cdeef152abc5d8a68122510R89), and would eliminate a couple of `#if`s.